### ISSUE TITLE
Default values for list fields

### ISSUE THREAD

**Issue by ItayGoren** [nazrulworld/fhir.resources#34]
* fhir.resources version: 6.0.0
* Python version: 3.8
* Operating System: MacOS

### Description

I noticed that all the optional objects which has amount of `0...1` are defaulted to `None` which is logical.
Why do we set the default value for fields with amount `0...*` to be `None` as well and not `[]`?

**Comment by mmabey** [comment]
**TL;DR**

What you're suggesting is already taken care of automatically by Pydantic.

**Verbose Explanation**

This really comes down to how Pydantic uses the "default" value. If you take a look at the [docstring for that field](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/fields.py#L146), you'll notice that what that parameter is actually allowing you to do is *replace* the default value, not set it.

For example, look at the [definition](https://github.com/nazrulworld/fhir.resources/blob/master/fhir/resources/claim.py#L52) of the `careTeam` field of a `Claim`:

```python
careTeam: ListType[fhirtypes.ClaimCareTeamType] = Field(
    None,
    alias="careTeam",
    title="Members of the care team",
    description="The members of the team who provided the products and services.",
    # if property is element of this resource.
    element_property=True,
)
```

Because `careTeam` is a `ListType` (which is an alias for the Python `typing.List` type), Pydantic will *automatically* assign it a default value of `[]`. The [cardinality of this field](https://www.hl7.org/fhir/claim.html#resource) is of course `0..*` in the FHIR spec, meaning it is optional but can also have many values.

If we were to change the `default` parameter to `Field()` from `None` to `[]`, that would tell Pydantic to *overwrite* the default from `[]` to `[]`, which doesn't accomplish much.

And just to round out this discussion, for fields with cardinality `1..1` (like the `created` field of the `Claim` FHIR type), we pass the ellipsis (`...`) to tell Pydantic that we don't want to overwrite the default value, but there *must* be some value present.

**Comment by ItayGoren** [comment]
Look at:
```python
from fhir.resources.patient import Patient
list(Patient.element_properties())
```
it prints:
```
....
ModelField(name='name', type=Optional[List[HumanNameType]], required=False, default=None)
...
```
Yet, the field itself has the declaration:
```python
name: ListType[fhirtypes.HumanNameType] = Field(
```
And calling `print(Patient().name)` gives `None`.
Hence, what you are saying is not what actually happen.
